packages/lesswrong/components/users/DialogueMatchingPage.tsx

```diff
@@ -480,56 +805,77 @@ const DialogueCheckBox: React.FC<{
   );
 };
 
-const isMatched = (userDialogueChecks: DialogueCheckInfo[], targetUserId: string): boolean => {
-  return userDialogueChecks.some(check => check.targetUserId === targetUserId && check.match);
-};
-
-const isChecked = (userDialogueChecks: DialogueCheckInfo[], targetUserId: string): boolean => {
-  return userDialogueChecks?.find(check => check.targetUserId === targetUserId)?.checked || false;
-};
-
-type MatchDialogueButtonProps = {
-  isMatched: boolean;
-  targetUserId: string;
-  targetUserDisplayName: string;
-  currentUser: UsersCurrent;
-  loadingNewDialogue: boolean;
-  createDialogue: (title: string, participants: string[]) => void;
-  classes: ClassesType;
-};
-
 const MatchDialogueButton: React.FC<MatchDialogueButtonProps> = ({
   isMatched,
+  checkId,
   targetUserId,
   targetUserDisplayName,
   currentUser,
-  loadingNewDialogue,
-  createDialogue,
   classes,
 }) => {
+
+  const { openDialog } = useDialog();
+  const { history } = useNavigation();
+
+  const {loading: userLoading, results} = useMulti({
+    terms: {
+      view: "dialogueMatchPreferences",
+      dialogueCheckId: checkId,
+      limit: 1000,
+    },
+    fragmentName: "DialogueMatchPreferenceInfo",
+    collectionName: "DialogueMatchPreferences",
+  });
+
   if (!isMatched) return <div></div>; // need this instead of null to keep the table columns aligned
 
-  return (
-    <div>
+  const userMatchPreferences = results && results.length > 0 ? results[0] : null;
+  const generatedDialogueId = userMatchPreferences?.generatedDialogueId;
+
+  const renderButton = () => {
+    if (!!generatedDialogueId) {
+      return (
+        <button className={classes.lightGreenButton} onClick={(e) => redirect(generatedDialogueId, history)}>
+          <a data-cy="message">Go to dialogue</a>
+        </button>
+      );
+    }
+  
+    if (userMatchPreferences) {
+      return (
+        <button className={classes.waitingButton} disabled>
+          <a data-cy="message">Waiting for {targetUserDisplayName}...</a>
+        </button>
+      );
+    }
+  
+    return (
       <button
-        className={classes.newDialogueButton}
+        className={classes.enterTopicsButton}
         onClick={(e) =>
-          createDialogue(
-            `${currentUser?.displayName}/${targetUserDisplayName}`,
-            [targetUserId]
-          )
+          openDialog({
+            componentName: 'NextStepsDialog',
+            componentProps: {
+              userId: currentUser?._id,
+              targetUserId,
+              targetUserDisplayName,
+              dialogueCheckId: checkId!
```

Comment:
Why non-null assertion here? Please avoid and find otherwise to satisfy type-check. Given the nature of the component, `checkId` should never be null, so maybe check over your types for the component.